src/main/java/de/tum/in/www1/artemis/service/notifications/GroupNotificationService.java

```diff
@@ -57,12 +57,19 @@ public GroupNotificationService(GroupNotificationRepository groupNotificationRep
      * @param notificationText that is used for the notification process
      */
     public void notifyAboutExerciseUpdate(Exercise exercise, String notificationText) {
+
+        if (exercise.isExamExercise()) {
+            // Do not send an exercise-update notification if it's an exam exercise.
+            // Exam exercise updates are handled using exam live events.
+            return;
+        }
+
         if (exercise.getReleaseDate() != null && exercise.getReleaseDate().isAfter(ZonedDateTime.now())) {
             // Do not send an exercise-update notification before the release date of the exercise.
             return;
         }
 
-        if ((notificationText != null && exercise.isCourseExercise()) || exercise.isExamExercise()) {
+        if ((notificationText != null && exercise.isCourseExercise())) {
```

Comment:
This is always a course exercise thanks to the check above.
```suggestion
        if (notificationText != null) {
```

Reply:
good points, thank you!